apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/module-instance.tsx

```diff
@@ -0,0 +1,55 @@
+"use client";
+
+import { lazy } from "react";
+import type { ThirdwebContract } from "thirdweb";
+import type { Account } from "thirdweb/wallets";
+import BatchMetadataModule from "./BatchMetadata";
+import ClaimableModule from "./Claimable";
+import OpenEditionMetadataModule from "./OpenEditionMetadata";
+import { ModuleCardUI, type ModuleCardUIProps } from "./module-card";
+
+const MintableModule = lazy(() => import("./Mintable"));
+const TransferableModule = lazy(() => import("./Transferable"));
+const RoyaltyModule = lazy(() => import("./Royalty"));
+
+export type ModuleInstanceProps = Omit<
+  ModuleCardUIProps,
+  "children" | "updateButton" | "isOwnerAccount"
+> & {
+  contract: ThirdwebContract;
+  ownerAccount: Account | undefined;
+  allModuleContractInfo: {
+    name: string;
+    description?: string;
+    version?: string;
+    publisher?: string;
+  }[];
+};
+
+export function ModuleInstance(props: ModuleInstanceProps) {
+  if (props.contractInfo.name.includes("Transferable")) {
+    return <TransferableModule {...props} />;
+  }
+
+  if (props.contractInfo.name.includes("Mintable")) {
```

Comment:
Is checking `name` property the best way to determine the module type?

Reply:
For now it is, that's the most consistent piece of information we currently have